### PR TITLE
bring evalset up to date with internal test_funcs.py

### PR DESCRIPTION
**Pull request by samuela**
re https://app.asana.com/0/38684300131201/192137498275988
cc @iandewancker @mikemccourt @pfhayes

### CODE REVIEW

**Comment by mikemccourt** [comment]
Is this all the function changes?  Somehow I had thought there would be more (although alot were done prior to the evalset being released so maybe I'm remembering some of that).

**Comment by samuela** [comment]
I was also surprised. But this brings us up-to-date with the version of `test_funcs.py` in eval_framework on master.

**Comment by mikemccourt** [comment]
Okay, sounds good.  We should have a function that also, through
super-duper brute force, can be run to test and make sure that the max
(worst) and min (best) function vals stored in the functions is
accurate-ish.  Don't need that right now, though.

**Comment by samuela** [comment]
@mikemccourt I suppose there are a few ways we could test the functions:
- Test that the value at `min_loc` is `fmin`. (cheap)
- Test randomly sampled points in the domain and check that they are all within `[fmin, fmax]`. (not as cheap)
- Use scipy.optimize to exhaustively identify the min/max and check that they are roughly equal. (expensive)

**Comment by mikemccourt** [comment]
Don't worry about that now.  I'll worry about that some other time.  Just
something I had in my notes to look at.  I usually use the scipy.optimize
to test functions I want to put in eval so that would be the option I would
use.

File: test_functions/test_funcs.py [hunk 1]
```diff
@@ -99,28 +101,42 @@ class TestFunction(object):
...
-        x = numpy.asarray(x)
-        if x.shape != (self.dim,):
-          raise ValueError('Argument must be a numpy array of length {}'.format(self.dim))
+        if self.verify and (not isinstance(x, numpy.ndarray) or x.shape != (self.dim,)):
+            raise ValueError('Argument must be a numpy array of length {}'.format(self.dim))
 
         self.num_evals += 1
         value = self.do_evaluate(x)
+        to_be_returned = value.item() if hasattr(value, 'item') else value
+        self.update_records(now(), x, to_be_returned)
```
Comment (pfhayes) [inline test_functions/test_funcs.py:129]: why are we using `timeit` for this? why not just `time.time()` or `datetime.now()`?

File: test_functions/test_funcs.py [hunk 2]
```diff
@@ -49,8 +49,10 @@
 
 from scipy.special import jv as besselj
 
-seterr(all='ignore')
+from timeit import default_timer as now
+from collections import namedtuple
```
Comment (pfhayes) [inline test_functions/test_funcs.py:53]: is this used?

File: test_functions/test_funcs.py [hunk 3]
```diff
@@ -49,8 +49,10 @@
...
 
+seterr(all='ignore')
```
Comment (pfhayes) [inline test_functions/test_funcs.py:55]: what is this for?

File: test_functions/test_funcs.py (hunk 2 above)
Comment (samuela) [reply test_functions/test_funcs.py:53]: Hopefully not...

**Comment by samuela** [comment]
@pfhayes Not sure that I can speak to either of these questions. This code has only been copied from the eval_framework version, so I'm not sure what the original intentions were.

**Comment by pfhayes** [comment]
Ok, lgtm

**Comment by mikemccourt** [comment]
Yeah, those things are used in eval (except the seterr thing) but not used
in just this function.  This will be one of the pains of syncing this crap
up.

As far as timeit - just what I always use, feel free to change it.
 namedtuple can go, that should have been gone a whlie ago when that got
pushed down to eval_test_case.

**Comment by samuela** [comment]
Is it necessary to include this eval_framework specific stuff in `test_funcs.py`?

**Comment by mikemccourt** [comment]
Maybe not, but it's there now.

**Comment by samuela** [comment]
Merging for now and we can clean up some of these eval_framework vestiges as we go forward.

**Pull request merged**